library/core/src/main/java/com/google/android/exoplayer2/audio/DefaultAudioSink.java

```diff
@@ -342,15 +365,20 @@ public void configure(@C.Encoding int inputEncoding, int inputChannelCount, int
     int channelCount = inputChannelCount;
     int sampleRate = inputSampleRate;
     isInputPcm = isEncodingPcm(inputEncoding);
+    shouldUpResPCMAudio = canConvertHiResPcmToFloat &&
+     (inputEncoding == C.ENCODING_PCM_24BIT || inputEncoding == C.ENCODING_PCM_32BIT);
     if (isInputPcm) {
-      pcmFrameSize = Util.getPcmFrameSize(inputEncoding, channelCount);
+      pcmFrameSize = Util.getPcmFrameSize(shouldUpResPCMAudio
```

Comment:
pcmFrameSize is used to calculate the number of submitted frames (not the number of output frames), so shouldn't this be left as it was?

Reply:
Yes, I think that was a mixing of changes that got left in and caused discontinuity errors.  Fixed.